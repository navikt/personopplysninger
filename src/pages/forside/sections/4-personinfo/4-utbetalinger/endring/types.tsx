import { OptionType } from '../../../../../../types/option';

export interface OutboundNorskKontonummer {
    value: string;
}

export interface OutboundUtenlandsbankonto {
    value: string | null;
    utenlandskKontoInformasjon: {
        landkode: string;
        landkodeTobokstavs?: string;
        valuta: string;
        swift?: string;
        bank: {
            adresseLinje1?: string;
            adresseLinje2?: string;
            adresseLinje3?: string;
            kode?: string;
            navn: string;
        };
    };
}

export interface FormFields {
    kontonummer: string;
    land?: OptionType;
    valuta?: OptionType;
    banknavn: string;
    kontonummerIban: string;
    bickode: string;
    retningsnummer: string;
    bankkode: string;
    adresse1: string;
    adresse2: string;
    adresse3: string;
}
