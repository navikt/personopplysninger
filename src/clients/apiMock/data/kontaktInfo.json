{
  "mobiltelefonnummer": "12131415",
  "epostadresse": "test@test.com",
  "kanVarsles": true
}
