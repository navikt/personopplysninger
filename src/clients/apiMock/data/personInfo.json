{
  "personalia": {
    "fornavn": "DONALD FAUNTLEROY",
    "etternavn": "DUCK",
    "personident": {
      "verdi": "12345678911",
      "type": "FNR"
    },
    "utenlandskbank": {
      "adresse1": "Gävlegatan 22",
      "adresse2": "113 30 Stockholm",
      "adresse3": "",
      "bankkode": "90081210188",
      "banknavn": "Nordax Bank AB",
      "iban": "",
      "kontonummer": "US90081210188",
      "land": "Sverige",
      "swiftkode": "HELGSE21",
      "valuta": "Tugriks Mongolia"
    },
    "tlfnr": {
      "jobb": "12121212",
      "mobil": "24242424",
      "privat": "36363636",
      "datakilder": [{}, {}]
    },
    "spraak": "NB",
    "epostadr": "TODO",
    "personstatus": "BOSA",
    "statsborgerskap": "NOR",
    "foedested": "NOR",
    "sivilstand": "UGIF",
    "kjoenn": "M",
    "datakilder": [{}, {}, {}, {}]
  },
  "enhetKontaktInformasjon": {
    "enhet": {
      "postadresse": {
        "type": "stedsadresse",
        "postnummer": "0614",
        "poststed": "Oslo",
        "gatenavn": "Slottsveien",
        "husnummer": "22",
        "husbokstav": "A"
      },
      "publikumsmottak": [
        {
          "gateadresse": "Peder Falcks veg",
          "poststed": "Trondheim",
          "husnummer": "31",
          "husbokstav": null,
          "postnummer": "7044",
          "stedsbeskrivelse": null,
          "aapningMandag": {
            "dag": "Mandag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": "Timeavtaler også utenom åpningstiden"
          },
          "aapningTirsdag": {
            "dag": "Tirsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningOnsdag": {
            "dag": "Onsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningTorsdag": {
            "dag": "Torsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningFredag": {
            "dag": "Fredag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningAndre": [
            {
              "dag": "15.05.2019",
              "fra": "09:00",
              "til": "12:00",
              "stengt": "false",
              "kommentar": "Her kan det også være en merknad "
            },
            {
              "dag": "17.05.2019",
              "fra": null,
              "til": null,
              "stengt": "true"
            },
            {
              "dag": "29.05.2019",
              "fra": "09:00",
              "til": "13:30",
              "stengt": "false",
              "kommentar": "Onsdag før Kristi himmelfartsdag"
            }
          ]
        },
        {
          "gateadresse": "Kirkegata",
          "poststed": "Trondheim",
          "husnummer": "25",
          "husbokstav": null,
          "postnummer": "7014",
          "stedsbeskrivelse": null,
          "aapningMandag": {
            "dag": "Mandag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": "Timeavtaler også utenom åpningstiden"
          },
          "aapningTirsdag": {
            "dag": "Tirsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningOnsdag": {
            "dag": "Onsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningTorsdag": {
            "dag": "Torsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningFredag": {
            "dag": "Fredag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningAndre": [
            {
              "dag": "15.05.2019",
              "fra": "09:00",
              "til": "12:00",
              "stengt": "false",
              "kommentar": "Her kan det også være en merknad "
            },
            {
              "dag": "17.05.2019",
              "fra": null,
              "til": null,
              "stengt": "true"
            },
            {
              "dag": "29.05.2019",
              "fra": "09:00",
              "til": "13:30",
              "stengt": "false",
              "kommentar": "Onsdag før Kristi himmelfartsdag"
            }
          ]
        }
      ],
      "tlfperson": "55553333",
      "spesielleopplysninger": "Timeavtale 8-15, man - fre."
    }
  },
  "adresser": {
    "boadresse": {
      "adressetillegg": "C/O Rune Aremark",
      "adresse": "Bjørvika 4D",
      "datoFraOgMed": "2017-02-21",
      "kommune": "1239",
      "land": "Norge",
      "matrikkeladresse": {},
      "postnummer": "1239",
      "poststed": "Jepp",
      "veiadresse": {
        "bokstav": "A",
        "bolignummer": "H0102",
        "gatekode": "01239",
        "husnummer": "11"
      }
    },
    "geografiskTilknytning": {
      "bydel": "500102",
      "datoFraOgMed": "2012-10-01",
      "kommune": null,
      "land": null,
      "enhet": "NAV Falkenborg"
    },
    "postadresse": {
      "adresse1": "12345 HFJEF IQUFJ HFURHGHQBEUQ",
      "adresse2": "BATOOM LUNCHAGRAMA,GONGRUPEA,DD",
      "adresse3": "GGLAM,10000 HCAMGHU,THAILAND",
      "datoFraOgMed": "2018-03-13",
      "land": "THA",
      "postnummer": "",
      "poststed": "Nittedal"
    },
    "prioritertAdresse": "TIAD",
    "tilleggsadresse": {
      "adresse1": "JEPP 10jfioew JIO LPFL ",
      "adresse2": "LOOOL FKEOWIF JEWIOF WECOOOO FJEI",
      "adresse3": "",
      "bolignummer": "",
      "bydel": "",
      "datoFraOgMed": "2019-01-12",
      "datoTilOgMed": "2023-01-11",
      "gateKode": "0270",
      "husbokstav": "",
      "husnummer": "0020",
      "kommunenummer": "0131",
      "postboksanlegg": "",
      "postboksnummer": "",
      "postnummer": "2021",
      "poststed": "JAJAJA"
    },
    "datakilder": [{}, {}]
  }
}
