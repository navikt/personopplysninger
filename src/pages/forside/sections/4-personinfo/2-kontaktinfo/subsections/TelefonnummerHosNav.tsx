import React from "react";
import { Undertittel } from "nav-frontend-typografi";
import { FormattedMessage } from "react-intl";
import ListElement from "../../../../../../components/listelement/ListElement";
import Melding from "../../../../../../components/melding/Melding";
import { Tlfnr } from "../../../../../../types/personalia";
import Kilde from "../../../../../../components/kilde/Kilde";
import Environment from "../../../../../../utils/Environments";
import endreIkon from "../../../../../../assets/img/Pencil.svg";
import leggTilIkon from "../../../../../../assets/img/LeggTil.svg";

const { tjenesteUrl } = Environment();

interface Props {
  tlfnr?: Tlfnr;
}

const TelefonnummerHosNav = (props: Props) => {
  const { tlfnr } = props;
  return (
    <>
      <div className="underseksjon__header">
        <Undertittel>
          <FormattedMessage id="personalia.tlfnr.oveskrift" />
        </Undertittel>
      </div>
      {tlfnr && (tlfnr.jobb || tlfnr.mobil || tlfnr.privat) ? (
        <>
          <ul className="list-column-2">
            <ListElement
              titleId="personalia.tlfnr.arbeid"
              content={tlfnr.jobb}
            />
            <ListElement
              titleId="personalia.tlfnr.mobil"
              content={tlfnr.mobil}
            />
            <ListElement
              titleId="personalia.tlfnr.hjem"
              content={tlfnr.privat}
            />
          </ul>
          <Kilde
            kilde="personalia.source.nav"
            lenke={`${tjenesteUrl}/brukerprofil/`}
            lenkeTekst="personalia.link.brukerprofil.endre"
            lenkeType={"EKSTERN"}
            ikon={endreIkon}
          />
        </>
      ) : (
        <>
          <Melding meldingId="personalia.tlfnr.ingenData" />
          <Kilde
            kilde="personalia.source.nav"
            lenke={`${tjenesteUrl}/brukerprofil/`}
            lenkeTekst="personalia.link.brukerprofil.leggtil"
            lenkeType={"EKSTERN"}
            ikon={leggTilIkon}
          />
        </>
      )}
    </>
  );
};

export default TelefonnummerHosNav;
