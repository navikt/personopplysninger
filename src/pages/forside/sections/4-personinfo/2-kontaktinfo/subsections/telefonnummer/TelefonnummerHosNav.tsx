import React, { useState } from "react";
import { FormattedMessage } from "react-intl";
import { Tlfnr } from "types/personalia";
import leggTilIkon from "assets/img/LeggTil.svg";
import Kilde from "components/kilde/Kilde";
import Telefonnummer from "./Telefonnummer";
import TelefonnummerForm from "./TelefonnummerForm";
import { fjernMellorom } from "utils/formattering";
import driftsmeldinger from "driftsmeldinger";
import { Alert, Button, Label } from "@navikt/ds-react";

interface Props {
  tlfnr?: Tlfnr;
}

const TelefonnummerHosNav = (props: Props) => {
  const [opprett, settOpprett] = useState<boolean>();
  const { tlfnr } = props;

  const onChangeSuccess = () => {
    settOpprett(false);
  };

  const onDeleteSuccess = () => {
    settOpprett(false);
  };

  const onLeggTil = () => settOpprett(!opprett);

  return (
    <>
      {driftsmeldinger.pdl && (
        <div style={{ paddingBottom: "1rem" }}>
          <Alert variant="warning">{driftsmeldinger.pdl}</Alert>
        </div>
      )}
      {tlfnr && (tlfnr.telefonHoved || tlfnr.telefonAlternativ) ? (
        <div>
          {tlfnr.telefonHoved && (
            <Telefonnummer
              prioritet={1}
              titleId="personalia.tlfnr.telefon"
              hasTwoNumbers={!!(tlfnr.telefonHoved && tlfnr.telefonAlternativ)}
              landskode={tlfnr.landskodeHoved}
              tlfnummer={fjernMellorom(tlfnr.telefonHoved)}
              onDeleteSuccess={onDeleteSuccess}
              onChangeSuccess={onChangeSuccess}
            />
          )}
          {tlfnr.telefonAlternativ && (
            <Telefonnummer
              prioritet={2}
              titleId="personalia.tlfnr.telefon"
              hasTwoNumbers={!!(tlfnr.telefonHoved && tlfnr.telefonAlternativ)}
              landskode={tlfnr.landskodeAlternativ}
              tlfnummer={fjernMellorom(tlfnr.telefonAlternativ)}
              onDeleteSuccess={onDeleteSuccess}
              onChangeSuccess={onChangeSuccess}
            />
          )}
          {<Kilde kilde="personalia.source.nav" lenkeType={"INGEN"} />}
        </div>
      ) : (
        <div className="underseksjon__beskrivelse">
          <FormattedMessage
            id="personalia.tlfnr.ingenData"
            values={{
              b: (text) => <b>{text}</b>,
              br: (text) => (
                <>
                  <br />
                  {text}
                </>
              ),
            }}
          />
        </div>
      )}

      {!opprett &&
        !(tlfnr && tlfnr.telefonHoved && tlfnr.telefonAlternativ) && (
          <Button
            variant="tertiary"
            onClick={onLeggTil}
            className="tlfnummer__leggtil knapp-med-ikon lenke"
            aria-label="Legg til telefonnummer i NAV sitt register"
          >
            <img className="kilde__icon" src={leggTilIkon} alt="" />
            <FormattedMessage id={"side.leggtil.kontaktinformasjon"} />
          </Button>
        )}

      {opprett && (
        <div className={"tlfnummer__rad-leggtil"}>
          <div className={"tlfnummer__container"}>
            <Label className="tlfnummer__verdi" as="p">
              <FormattedMessage id="side.leggtil.kontaktinformasjon" />
            </Label>
          </div>
          <TelefonnummerForm
            type={"opprett"}
            prioritet={tlfnr && tlfnr.telefonHoved ? 2 : 1}
            onCancelClick={() => settOpprett(false)}
            onChangeSuccess={onChangeSuccess}
            tlfnr={tlfnr}
            defaultValues={{
              landskode: {
                label: "Norge",
                value: "+47",
              },
              tlfnummer: "",
            }}
          />
        </div>
      )}
    </>
  );
};

export default TelefonnummerHosNav;
