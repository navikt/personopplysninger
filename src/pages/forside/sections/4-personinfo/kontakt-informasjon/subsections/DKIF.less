.dkif {
  &__overskrift-container {
    display: flex;
    align-items: center;
  }
  &__overskrift-i-icon {
    max-width: 1.5rem;
    min-width: 1.5rem;
    margin: 0 1rem;
    cursor: pointer;
  }
}
