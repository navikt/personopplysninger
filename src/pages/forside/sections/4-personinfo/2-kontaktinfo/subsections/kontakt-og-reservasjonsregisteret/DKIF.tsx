import React from "react";
import { FormattedMessage } from "react-intl";
import ListElement from "components/listelement/ListElement";
import { KontaktInfo } from "types/kontaktInfo";
import Kilde from "components/kilde/Kilde";
import eksternLenkeIkon from "assets/img/Link.svg";
import { useStore } from "../../../../../../../store/Context";

interface Props {
  info: KontaktInfo;
}

const KontaktInformasjon = (props: Props) => {
  const { mobiltelefonnummer, epostadresse, kanVarsles, spraak } = props.info;
  const [{ locale }] = useStore();
  return mobiltelefonnummer || epostadresse || kanVarsles ? (
    <>
      <dl className="list">
        <ListElement titleId="personalia.tlfnr" content={mobiltelefonnummer} />
      </dl>
      <dl className="list">
        <ListElement titleId="personalia.email" content={epostadresse} />
      </dl>
      <dl className="list">
        <ListElement titleId="personalia.spraak" content={spraak} />
      </dl>
      <Kilde
        kilde="personalia.source.dkif"
        lenke={`https://brukerprofil.difi.no/minprofil${
          locale === "en" ? "?locale=en" : ""
        }`}
        lenkeTekst="personalia.link.dkif.endre"
        lenkeType={"EKSTERN"}
        ikon={eksternLenkeIkon}
      />
    </>
  ) : (
    <>
      <div className="underseksjon__beskrivelse">
        <FormattedMessage
          id={"personalia.dkif.ingenData"}
          values={{
            b: (text) => <b>{text}</b>,
          }}
        />
      </div>
      <Kilde
        kilde="personalia.source.dkif"
        lenke="https://brukerprofil.difi.no/minprofil"
        lenkeTekst="personalia.link.dkif.leggtil"
        lenkeType={"EKSTERN"}
        ikon={eksternLenkeIkon}
      />
    </>
  );
};

export default KontaktInformasjon;
