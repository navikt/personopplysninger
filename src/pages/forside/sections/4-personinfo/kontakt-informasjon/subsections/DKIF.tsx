import React, { useEffect } from "react";
import { Undertittel } from "nav-frontend-typografi";
import { FormattedMessage } from "react-intl";
import Error, { HTTPError } from "../../../../../../components/error/Error";
import Spinner from "../../../../../../components/spinner/Spinner";
import { fetchKontaktInfo } from "../../../../../../clients/apiClient";
import { KontaktInfo } from "../../../../../../types/kontaktInfo";
import KontaktInformasjon from "./KontaktInformasjon";
import { useStore } from "../../../../../../providers/Provider";
import Infotekst from "../../../../../../components/infotekst/Infotekst";
import { injectIntl, InjectedIntlProps } from "react-intl";

export type FetchKontaktInfo =
  | { status: "LOADING" }
  | { status: "RESULT"; data: KontaktInfo }
  | { status: "ERROR"; error: HTTPError };

const DKIF = (props: InjectedIntlProps) => {
  const [{ kontaktInfo }, dispatch] = useStore();

  useEffect(() => {
    if (kontaktInfo.status === "LOADING") {
      fetchKontaktInfo()
        .then(kontaktInfo =>
          dispatch({
            type: "SETT_KONTAKT_INFO_RESULT",
            payload: kontaktInfo as KontaktInfo
          })
        )
        .catch((error: HTTPError) =>
          dispatch({ type: "SETT_KONTAKT_INFO_ERROR", payload: error })
        );
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <>
      <hr className="box__linje-bred" />
      <div className="underseksjon__overskrift">
        <div className="dkif__overskrift-container">
          <Undertittel>
            <FormattedMessage id="personalia.dkif.overskrift" />
          </Undertittel>
          <Infotekst
            beskrivelse={props.intl.formatMessage({
              id: "personalia.dkif.beskrivelse"
            })}
          />
        </div>
      </div>
      {(() => {
        switch (kontaktInfo.status) {
          case "LOADING":
            return <Spinner />;
          case "RESULT":
            return <KontaktInformasjon info={kontaktInfo.data} />;
          case "ERROR":
            return <Error error={kontaktInfo.error} />;
        }
      })()}
    </>
  );
};

export default injectIntl(DKIF);
