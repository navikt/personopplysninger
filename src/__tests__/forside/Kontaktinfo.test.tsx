import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import nbMessages from '@/text/nb';
import kontaktInfo from '@/clients/apiMock/app/fetch/kontakt-info.json';
import personInfo from '@/clients/apiMock/app/fetch/person-info.json';
import DKIF from '@/pages/forside/sections/4-personinfo/2-kontaktinfo/subsections/kontakt-og-reservasjonsregisteret/DKIF';
import TelefonnummerHosNav from '@/pages/forside/sections/4-personinfo/2-kontaktinfo/subsections/telefonnummer/TelefonnummerHosNav';
import { Tlfnr } from '@/types/personalia';
import { StoreProvider } from '@/store/Context';

describe('DKIF', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <StoreProvider>
                <IntlProvider locale={'nb'} messages={nbMessages}>
                    <DKIF info={kontaktInfo} />
                </IntlProvider>
            </StoreProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('TelefonnummerHosNav', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <StoreProvider>
                <IntlProvider locale={'nb'} messages={nbMessages}>
                    <TelefonnummerHosNav tlfnr={personInfo.personalia.tlfnr as unknown as Tlfnr} />
                </IntlProvider>
            </StoreProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
