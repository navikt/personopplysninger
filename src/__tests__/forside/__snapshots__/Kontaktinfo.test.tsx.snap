// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DKIF renders correctly 1`] = `
<DocumentFragment>
  <dl
    class="list"
  >
    <div
      class="list-item"
    >
      <dt
        class="list-element-heading"
      >
        <span
          class="navds-label"
        >
          Telefon
        </span>
      </dt>
      <dd>
        12131415
      </dd>
    </div>
  </dl>
  <dl
    class="list"
  >
    <div
      class="list-item"
    >
      <dt
        class="list-element-heading"
      >
        <span
          class="navds-label"
        >
          E-post
        </span>
      </dt>
      <dd>
        test@test.com
      </dd>
    </div>
  </dl>
  <dl
    class="list"
  >
    <div
      class="list-item"
    >
      <dt
        class="list-element-heading"
      >
        <span
          class="navds-label"
        >
          Språk/målform
        </span>
      </dt>
      <dd>
        Norsk
      </dd>
    </div>
  </dl>
  <div
    class="kilde__container"
  >
    <p
      class="navds-body-long"
    >
      <a
        class="navds-link kilde__lenke lenke"
        href="https://brukerprofil.difi.no/minprofil"
      >
        <span
          class="kilde__icon"
        >
          <img
            alt=""
            src="Link.svg"
          />
        </span>
        Vil du endre informasjonen i Kontakt- og reservasjonsregisteret? 
      </a>
    </p>
    <div
      class="kilde__tekst"
    >
      <p
        class="navds-detail"
      >
        Kilde: Kontakt - og reservasjonsregisteret
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TelefonnummerHosNav renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="tlfnummer__rad"
    >
      <div
        class="tlfnummer__container"
      >
        <div
          class="tlfnummer__verdi"
        >
          <p
            class="navds-label"
          >
            Telefon 
          </p>
          <p
            class="navds-body-short"
          >
            <span>
              +47 
            </span>
            408 53 158
          </p>
        </div>
        <div
          class="tlfnummer__knapper"
        >
          <button
            aria-label="Endre telefonnummer"
            class="knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
          >
            <span
              aria-live="polite"
              class="navds-label"
            >
              <img
                alt=""
                class="tlfnummer__knapp-ikon"
                src="Pencil.svg"
              />
              <span
                class="tlfnummer__knapp-tekst"
              >
                Endre
              </span>
            </span>
          </button>
          <button
            aria-label="Slett telefonnummer"
            class="knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
          >
            <span
              aria-live="polite"
              class="navds-label"
            >
              <img
                alt=""
                class="tlfnummer__knapp-ikon"
                src="Slett.svg"
              />
              <span
                class="tlfnummer__knapp-tekst"
              >
                Slett
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <button
    aria-label="Legg til telefonnummer i NAV sitt register"
    class="tlfnummer__leggtil knapp-med-ikon lenke navds-button navds-button--tertiary navds-button--medium"
  >
    <span
      aria-live="polite"
      class="navds-label"
    >
      <img
        alt=""
        class="kilde__icon"
        src="LeggTil.svg"
      />
      Legg til nummer i NAV sitt register
    </span>
  </button>
  <div
    class="kilde__container"
  >
    <div
      class="kilde__tekst"
    >
      <p
        class="navds-detail"
      >
        Kilde: NAV
      </p>
    </div>
  </div>
</DocumentFragment>
`;
